File: flow/src/test/java/com/sequenceiq/cloudbreak/core/flow2/config/Flow2InitializerTest.java
package com.sequenceiq.cloudbreak.core.flow2.config;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.BDDMockito.given;
import static org.mockito.Mockito.times;
import static org.mockito.Mockito.verify;

import java.util.ArrayList;
import java.util.List;

import org.junit.Before;
import org.junit.Test;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import com.sequenceiq.cloudbreak.core.flow2.Flow2Handler;
import com.sequenceiq.cloudbreak.core.flow2.helloworld.HelloWorldFlowConfig;

import reactor.bus.EventBus;
import reactor.bus.selector.Selector;
import reactor.fn.Consumer;

public class Flow2InitializerTest {

    @InjectMocks
    private Flow2Initializer underTest;

    @Mock
    private List<FlowConfiguration<?>> flowConfigs;

    @Mock
    private EventBus reactor;

    @Mock
    private Flow2Handler flow2Handler;

    @Before
    public void setUp() {
        underTest = new Flow2Initializer();
        MockitoAnnotations.initMocks(this);
    }

    @Test
    public void testInitialize() {
        List<FlowConfiguration<?>> flowConfigs = new ArrayList<>();
        flowConfigs.add(new HelloWorldFlowConfig());
        given(this.flowConfigs.stream()).willReturn(flowConfigs.stream());
        underTest.init();
        verify(reactor, times(1)).on(any(Selector.class), any(Consumer.class));
    }
}

File: core/src/test/java/com/sequenceiq/cloudbreak/core/flow2/config/Flow2InitializerTest.kt
package com.sequenceiq.cloudbreak.core.flow2.config

import org.mockito.BDDMockito.given
import org.mockito.Matchers.any
import org.mockito.Mockito.times
import org.mockito.Mockito.verify

import java.util.ArrayList

import org.junit.Before
import org.junit.Test
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.MockitoAnnotations

import com.sequenceiq.cloudbreak.core.flow2.Flow2Handler
import com.sequenceiq.cloudbreak.core.flow2.stack.sync.StackSyncFlowConfig
import com.sequenceiq.cloudbreak.core.flow2.stack.termination.StackTerminationFlowConfig

import reactor.bus.EventBus
import reactor.bus.selector.Selector
import reactor.fn.Consumer

class Flow2InitializerTest {

    @InjectMocks
    private var underTest: Flow2Initializer? = null

    @Mock
    private val flowConfigs: List<FlowConfiguration<*>>? = null

    @Mock
    private val reactor: EventBus? = null

    @Mock
    private val flow2Handler: Flow2Handler? = null

    @Before
    fun setUp() {
        underTest = Flow2Initializer()
        MockitoAnnotations.initMocks(this)
    }

    @Test
    fun testInitialize() {
        val flowConfigs = ArrayList<FlowConfiguration<*>>()
        flowConfigs.add(StackSyncFlowConfig())
        flowConfigs.add(StackTerminationFlowConfig())
        given<Stream<FlowConfiguration<*>>>(this.flowConfigs!!.stream()).willReturn(flowConfigs.stream())
        underTest!!.init()
        verify<EventBus>(reactor, times(1)).on<Event<*>>(any<Selector>(Selector<Any>::class.java), any<Consumer>(Consumer<Any>::class.java))
    }
}
